tests/Tokenizer/TokensAnalyzerTest.php

```diff
@@ -769,247 +769,236 @@ public function testIsLambdaInvalid()
+            [3 => false],
+        ];
+        yield [
+            '<?php use FOO;',
+            [3 => false],
+        ];
+        yield [
+            '<?php use function FOO\BAR\BAZ;',
+            [5 => false, 7 => false, 9 => false],
+        ];
+        yield [
+            '<?php namespace X; const FOO = 1;',
+            [3 => false, 8 => false],
+        ];
+        [
+            '<?php class FOO {}',
+            [3 => false],
+        ];
+        yield [
+            '<?php interface FOO {}',
+            [3 => false],
+        ];
+        yield [
+            '<?php trait FOO {}',
+            [3 => false],
+        ];
+        yield [
+            '<?php class x extends FOO {}',
+            [3 => false, 7 => false],
+        ];
+        yield [
+            '<?php class x implements FOO {}',
+            [3 => false, 7 => false],
+        ];
+        yield [
+            '<?php class x implements FOO, BAR, BAZ {}',
+            [3 => false, 7 => false, 10 => false, 13 => false],
+        ];
+        yield [
+            '<?php class x { const FOO = 1; }',
+            [3 => false, 9 => false],
+        ];
+        yield [
+            '<?php class x { use FOO; }',
+            [3 => false, 9 => false],
+        ];
+        yield [
+            '<?php class x { use FOO, BAR { FOO::BAZ insteadof BAR; } }',
+            [3 => false, 9 => false, 12 => false, 16 => false, 18 => false, 22 => false],
+        ];
+        yield [
+            '<?php function x (FOO $foo, BAR &$bar, BAZ ...$baz) {}',
+            [3 => false, 6 => false, 11 => false, 17 => false],
+        ];
+        yield [
+            '<?php FOO();',
+            [1 => false],
+        ];
+        yield [
+            '<?php FOO::x();',
+            [1 => false, 3 => false],
+        ];
+        yield [
+            '<?php x::FOO();',
+            [1 => false, 3 => false],
+        ];
+        yield [
+            '<?php $foo instanceof FOO;',
+            [5 => false],
+        ];
+        yield [
+            '<?php try {} catch (FOO $e) {}',
+            [9 => false],
+        ];
+        yield [
+            '<?php "$foo[BAR]";',
+            [4 => false],
+        ];
+        yield [
+            '<?php "{$foo[BAR]}";',
+            [5 => true],
+        ];
+        yield [
+            '<?php FOO: goto FOO;',
+            [1 => false, 6 => false],
+        ];
+        yield [
+            '<?php foo(E_USER_DEPRECATED | E_DEPRECATED);',
+            [1 => false, 3 => true, 7 => true],
+        ];
+        yield [
+            '<?php interface Foo extends Bar, Baz, Qux {}',
+            [3 => false, 7 => false, 10 => false, 13 => false],
+        ];
+        yield [
+            '<?php use Foo\Bar, Foo\Baz, Foo\Qux;',
+            [3 => false, 5 => false, 8 => false, 10 => false, 13 => false, 15 => false],
         ];
-    }
-
-    /**
-     * @param string $source
-     *
-     * @dataProvider provideIsConstantInvocation70Cases
-     * @requires PHP 7.0
```

Comment:
:-1: 

with original code, we got nice `skipped test` message.
now, we don't even know that some test got skipped, as it was never generated

Reply:
please recover the original way, it will also save a lot of conflicts while merging 2.19 <> 3.0

Reply:
👍

Reply:
> with original code, we got nice `skipped test` message.

what is nice about those messages? Running on PHP 8 182 are skipped, do you track them and verify if that number changes? it was always unnecessary noise to me

My reasoning is that data provider should provide test cases, and there is nothing wrong if it provides ones filtered for given PHP version.

Reply:
I agree with @keradus here. I think it's better to have a feedback about skipped tests. Otherwise, if some are skipped for wrong reasons, you won't notice it. IMO test cases generation logic should be avoided as much as possible in data providers.

Reply:
Did anyone ever benefit from those skipped tests messages?

The tests might always be skipped for wrong reasons, skipping them over if conditions in data provider won't save from it.